examples/leaderregistry/main_test.go

```diff
@@ -498,6 +498,58 @@ func TestSurviveReplicaFailureWithRandomStrategy(t *testing.T) {
 	}
 }
 
+// TestRemoteServerTimeout tests the scenario where an actor is invoked with 2 replicas.
+// The invocation is done using a context with a deadline, and invoking a special operation that receives the expected
+// deadline as the payload. The actor operation checks whether the expected deadline matches the context deadline received by
+// the server. Due to HTTP lag, the test verifies that the deadlines are no more than 1 second apart.
```

Comment:
What do you think about propagating the timeout instead of the deadline to avoid this issue? I think thats what gRPC does: https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md